src/scaling/scaling.jl

```diff
@@ -0,0 +1,87 @@
+type ScaledInterpolation{T,N,ITPT,IT,GT,RT} <: AbstractInterpolationWrapper{T,N,ITPT,IT,GT}
+    itp::ITPT
+    ranges::RT
+end
+ScaledInterpolation{T,ITPT,IT,GT,RT}(::Type{T}, N, itp::ITPT, ::Type{IT}, ::Type{GT}, ranges::RT) =
+    ScaledInterpolation{T,N,ITPT,IT,GT,RT}(itp, ranges)
+@doc """
+`scale(itp, xs, ys, ...)` scales an existing interpolation object to allow for indexing using other coordinate axes than unit ranges, by wrapping the interpolation object and transforming the indices from the provided axes onto unit ranges upon indexing.
+
+The parameters `xs` etc must be either ranges or linspaces, and there must be one coordinate range/linspace for each dimension of the interpolation object.
+
+For every `NoInterp` dimension of the interpolation object, the range must be exactly `1:size(itp, d)`.
```

Comment:
Doesn't have to be part of this PR (or happen ever). But it's fine even for interpolating dimensions; after all, you let me supply `1:size(A,d)` for one of the knot vectors.